x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/DependencyConsistency.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.optimizer.rules;
+
+import org.elasticsearch.xpack.esql.common.Failures;
+import org.elasticsearch.xpack.esql.core.expression.Attribute;
+import org.elasticsearch.xpack.esql.core.expression.AttributeSet;
+import org.elasticsearch.xpack.esql.core.expression.NameId;
+import org.elasticsearch.xpack.esql.plan.QueryPlan;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.elasticsearch.xpack.esql.common.Failure.fail;
+
+public class DependencyConsistency<P extends QueryPlan<P>> {
```

Comment:
The name should indicate the role of this class - checking the consistency of dependencies. Right now it sounds like a DTO class which contains the dependencies consistency.

How about something more mundane but clearer such as `DependencyChecker`, `PlanStateChecker` or `ValidatePlanState`?

Reply:
This class was just extracted from an already existent one, but it wouldn't hurt a rename tbh.

Thinking about the name, I realized we don't have a javadoc for it to explain its intention. How about adding a sentence there?

Leaving the javadoc aside and just looking at the code, I am throwing another name suggestion in the bucket, but **up to @alex-spies to make the change or not altogether**: `PlanConsistencyChecker` (the reasoning behind the name is that this is a checker class and does something, an action; it looks at the "inputs" and "outputs" of each node plan if there is an inconsistency there it spits out an error, thus the `PlanConsistency` part). My 2c.

Reply:
I went with Andrei's suggestion :)